sites/game-client/map-renderer.mjs

```diff
@@ -139,6 +139,44 @@ class MapRenderer {
     };
   }
 
+  /**
+   * Determine whether the target is inside the viewport.
+   * @param {string} mapName - The map the client using.
+   * @param {object} canvasCoord1 - The target's left-top coordinate (x, y) relative to canvas.
+   * @param {object} canvasCoord2 - The target's right-bottom coordinate (x, y) relative to canvas.
+   * @return {bool} isInside - return true if the target is inside the viewport.
+   */
+  insideViewport(mapName, canvasCoord1, canvasCoord2) {
+    if (this.gameClient.playerInfo.mapCoord.mapName != mapName) {
```

Comment:
`if (this.viewerPosition.mapName === mapName)`

Reply:
You mean "!==" ?